### ISSUE TITLE
[BUG] New types of materials turn black after downloading save from SC-InteractiveMap

### ISSUE THREAD

**Issue by andremscunh** [AnthorNet/SC-InteractiveMap#121]
After I downloaded a modified save file from SC-InteractiveMap, I couldn't chage the color of any new materials added to the game. Is my save file doomed?

**Comment by AnthorNet** [comment]
Ok, before I can test and fix, i believe you need to lock again the resource sink bonus program at tier 2 and unlock it in game.

**Comment by andremscunh** [comment]
I think I found the actual problem, I can still change colors of materials. The problem is that all the new materials got their color changed to black when i compiled the calculator save and when I try to place a new block they come painted black as well. I bet the default color got changed somehow.

Before:
![image](https://user-images.githubusercontent.com/93388072/139589500-16f1fda5-6ffc-483b-b72a-93b60915e162.png)

After:
![image](https://user-images.githubusercontent.com/93388072/139589573-0be3fef5-b92a-4698-9cef-9ae7b84cb018.png)

**Comment by AnthorNet** [comment]
Not sure if it's the correct fix, but you can try to reset the Concrete swatch to the default color in the map to check.

**Comment by budn8** [comment]
I also had this exact issue with my save (default swatch for concrete structures was set to black after loading a _CALCULATOR save and couldn't be changed in game), but I was able to get it fixed! 

> Not sure if it's the correct fix, but you can try to reset the Concrete swatch to the default color in the map to check.

This is partially what fixed it for me. It didn't exactly work for me because the map editor thought the default colors were already applied to my swatch, but in game they were still black (see screenshots below). So clicking "Set to default" didn't do anything. 

![Black swatch in game](https://user-images.githubusercontent.com/39417521/140014197-b08f22db-6fae-46eb-a9cd-e2be403d2027.jpg)

![Interactive Map Swatches Menu](https://user-images.githubusercontent.com/39417521/140014204-5f5114d9-6268-406c-9578-18415b56f945.PNG)

__What finally fixed the issue is I changed the default concrete swatch using the Interactive Map, but the colors had to be changed to something completely different__ in order to fix the black swatch bug. I just picked two random colors that weren't the default white and orange. When I loaded back in with the new _CALCULATOR save, my new swatch was there and everything was working as expected. 🎉 

I wasn't able to reproduce this bug with different saves. My only guess is I believe I was using an outdated version of the map editor when the bug first occurred, so maybe that had something to do with it?
